File: StockExchange.DataContext/Entities/Order.cs
namespace StockExchange.DataContext.Entities
{
    using StockExchange.DataContext.Abstractions.Entities;
    public class Order : IIdentifiedEntity<Guid>
    {
        public Guid Id { get; private set; }
        public Guid InvestmentAccountId { get; private set; }
        public Guid StockId { get; private set; }
        public virtual InvestmentAccount? InvestmentAccount { get; private set; }
        public virtual Stock? Stock { get; private set; }
        public int NumberOfShares { get; private set; }
        public decimal TotalCost { get; private set; }

        public Order(Guid accountId, Guid stockId, int numberOfShares, decimal totalCost, Guid? id = null)
        {
            Id = id ?? Guid.NewGuid();
            InvestmentAccountId = accountId;
            StockId = stockId;
            NumberOfShares = numberOfShares;
            TotalCost = totalCost;
        }

        protected Order() { }
    }
}

File: stock-exchange-frontend/src/models/order.ts
export interface OrderPreview {
  stockId: string;
  numberOfShares: number;
  accountId: string;
}

export interface OrderCreate {
  accountId: string;
  stockId: string;
  numberOfShares: number;
}

export interface OrderPreviewModel {
  stockId: string;
  stockName: string;
  numberOfShares: number;
  totalCost: number;
  isOrderValid: boolean;
}

